segmentstore/contracts/src/main/java/io/pravega/segmentstore/contracts/StreamSegmentStore.java

```diff
@@ -114,6 +114,7 @@
      * consume the read data. If the operation failed, the future will be failed with the causing exception.
      * @throws NullPointerException     If any of the arguments are null.
      * @throws IllegalArgumentException If any of the arguments are invalid.
+     * @throws java.util.concurrent.CancellationException If the segment container is shutting down or the segment is evicted from memory.
```

Comment:
This is incorrect. The `read` method itself does not throw this `CancellationException`. The CompletableFuture that is returned may be cancelled, in which case it will complete exceptionally with such an exception.

Please remove this line. You may add a blurb in the `@return` tag, however I find that redundant, since every `CompletableFuture` out there may be cancelled.

Reply:
ok, updated the `@return` tag of the javadoc.